pkg/git/repository.go

```diff
@@ -252,6 +257,96 @@ func (o *CloneOptions) GetRepo(ctx context.Context) (Repository, fs.FS, error) {
 	return r, fs.Create(bootstrapFS), nil
 }
 
+var validateRepoWritePermission = func(ctx context.Context, r *repo) error {
+	cert, err := r.auth.GetCertificate()
+	if err != nil {
+		return fmt.Errorf("failed getting repository certificates: %w", err)
+	}
+
+	err = r.Repository.PushContext(ctx, &gg.PushOptions{
+		Auth:     getAuth(r.auth),
+		Progress: r.progress,
+		CABundle: cert,
+	})
+	if err != nil {
+		if !errors.Is(err, gg.NoErrAlreadyUpToDate) {
```

Comment:
If it is the 'already up to date' error , doesn't it mean you have write permissions?

Reply:
mm, actually didnt though about it, you are right

Reply:
actually surprising but no, you can have no write permission and you will get already up to date, but when try to push you will get error